src/poliastro/core/perturbations.py

```diff
@@ -204,6 +204,76 @@ def shadow_function(r_sat, r_sun, R):
     return theta > theta_1 + theta_2
 
 
+@jit
+def umbral_shadow(r_sat, r_sun, R):
```

Comment:
I think we should call this function `is_in_umbral_shadow`. Therefor, we could take the advantage in the `__call__` method of the `PenumbraEvent` of doing:

```python
return 0 if is_in_umbral_shadow(...) is True else 1
```

Notice that the `__call__` method returns a `0` if the event has happened or a different number if not.

Reply:
Hi @jorgepiloto I was thinking of the following:

Would `is_in_umbral_shadow` returning the difference between the vertical distances (as is done in the current commit) be a better choice instead of returning True or False? Since this would allow us to identify whether a satellite is going into or out of a particular shadow type (umbra or penumbra). Am I missing anywhere?

`r_sat_norm` is returned if it is not in any shadow (there we could put any non-zero value but I tried to choose `r_sat_norm` since it was already computed and that it would never be zero since its the distance from the center of the attractor and the satellite can only reach upto the attractor's radius).